lib/ansible/modules/cloud/pubnub/pubnub_blocks.py

```diff
@@ -0,0 +1,2373 @@
+                code = _content_of_file_at_path(src_path)
+                should_update = code and code != self.code
+            handler_state = _object_value(obj=event_handler, key='state', default='present')
+            should_update = should_update or handler_state == 'absent'
+        elif not should_update and self._service_data and self._event_handler_data:
+            should_update = not _is_equal(self._service_data, self._event_handler_data)
+
+        return should_update
+
+    def update_with_data(self, event_handler):
+        """Update handler data.
+
+        :type event_handler:  dict
+        :param event_handler: Reference on dictionary which represent event handler changes.
+        """
+        handler_state = _object_value(obj=event_handler, key='state', default='present')
+        if handler_state != 'absent':
+            # Retrieve user-provided handler information.
+            fields = ['name', 'src', 'channels', 'event']
+            (name, src_path, channels, event) = tuple(_values(dct=event_handler, keys=fields))
+            if _object_value(obj=event_handler, key='changes.name'):
+                name = event_handler['changes']['name']
+            self.name = name
+            if channels:
+                self.channels = channels
+            if event:
+                self.event = event
+            code = _content_of_file_at_path(src_path)
+            if code:
+                self.code = code
+        else:
+            self._should_delete = True
+
+    def save(self):
+        """Save event handler's changes.
+
+        Depending on whether event handler existed before or not it may be created and updated if
+        required.
+        """
+        will_change = self._should_create() or self.should_delete or self.will_change_with_data()
+        handler_data = self.payload
+        # Create new event handler if required.
+        if self._should_create():
+            fields = ['name', 'channels', 'event', 'code']
+            handler_data.update(self._default_handler_payload() or dict())
+            (name, channels, event, code) = _values(dct=handler_data, keys=fields)
+            if name and channels and event and code:
+                if not self._module.check_mode:
+                    response = self._api.create_event_handler(keyset_id=self._block.keyset.uid,
+                                                              payload=handler_data)
+                    handler_data.update(response or dict())
+                else:
+                    handler_data['id'] = 1617
+                self._changed = True
+            else:
+                missed_fields = list()
+                missed_fields.append('name') if not name else None
+                missed_fields.append('channels') if not channels else None
+                missed_fields.append('code') if not code else None
+                missed_fields.append('event') if not event else None
+                error_message = 'Un
```

Comment:
Typically the keywords for state we use are `started`, `stopped`, `present`, and `absent` since these are states rather than commands.